src/assessments/seo/textCompetingLinksAssessment.js

```diff
@@ -1,73 +1,116 @@
-};
 
-/**
- * Runs the linkCount module, based on this returns an assessment result with score.
- *
- * @param {object} paper The paper to use for the assessment.
- * @param {object} researcher The researcher used for calling research.
- * @param {object} i18n The object used for translations
- * @returns {object} the Assessmentresult
- */
-var textHasCompetingLinksAssessment = function( paper, researcher, i18n ) {
-	var linkCount = researcher.getResearch( "getLinkStatistics" );
+	/**
+	 * Runs the linkCount module, based on this returns an assessment result with score.
+	 *
+	 * @param {object} paper The paper to use for the assessment.
+	 * @param {object} researcher The researcher used for calling research.
+	 * @param {object} i18n The object used for translations
+	 * @returns {object} the AssessmentResult
+	 */
+	getResult( paper, researcher, i18n ) {
+		let assessmentResult = new AssessmentResult();
 
-	var linkCountResult = calculateLinkCountResult( linkCount, i18n );
-	var assessmentResult = new AssessmentResult();
+		this.linkCount = researcher.getResearch( "getLinkStatistics" );
 
-	assessmentResult.setScore( linkCountResult.score );
-	assessmentResult.setText( linkCountResult.text );
-	assessmentResult.setHasMarks( linkCountResult.hasMarks );
+		const calculatedResult = this.calculateResult();
+		if ( isUndefined( calculatedResult ) ) {
+			return assessmentResult;
+		}
 
-	return assessmentResult;
-};
+		assessmentResult.setScore( calculatedResult.score );
+		assessmentResult.setText( this.translateScore( calculatedResult.resultText, i18n ) );
+		assessmentResult.setHasMarks( true );
+		assessmentResult.setMarker( this.getMarks() );
 
-/**
- * Mark the anchors.
- *
- * @param {Paper} paper The paper to use for the marking.
- * @param {Researcher} researcher The researcher to use.
- * @returns {Array} Array with all the marked anchors.
- */
-var competingLinkMarker = function( paper, researcher ) {
-	var competingLinks = researcher.getResearch( "getLinkStatistics" );
+		return assessmentResult;
+	}
+
+	/**
+	 * Determines if the assessment is applicable to the paper.
+	 *
+	 * @param {Paper} paper The paper to check
+	 *
+	 * @returns {boolean} Whether the paper has text and a keyword
+	 */
+	isApplicable( paper ) {
+		return paper.hasText() && paper.hasKeyword();
+	}
+
+	/**
+	 * Returns a result based on the number of links.
+	 *
+	 * @returns {object} resultObject with score and text
+	 */
+	calculateResult() {
+		if ( this.linkCount.keyword.totalKeyword > this._config.parameters.recommendedMaximum ) {
+			return this._config.bad;
+		}
+	}
 
-	return map( competingLinks.keyword.matchedAnchors, function( matchedAnchor ) {
-		return new Mark( {
-			original: matchedAnchor,
-			marked: addMark( matchedAnchor ),
+	/**
+	 * Translates the score into a specific feedback to the user.
+	 *
+	 * @param {string} resultText The text string from the config to be returned for this number of occurrences of keyphrase
+	 * in the first paragraph.
```

Comment:
I think a shorter description such as `The feedback string` would also suffice.

Reply:
Fixed.